### PR TITLE
[ZEPPELIN-2596] Improving documentation page

### PR DESCRIPTION
**Pull request by 1ambda**
### What is this PR for?

Improving documentation page. Please check *TODO* and *Screenshots* sections for detail.
The motivation is described in [the JIRA ticket](https://issues.apache.org/jira/browse/ZEPPELIN-2583) and discussion is ongoing on the mailing list.

### What type of PR is it?
[Improvement | Documentation]

### Todos
* [x] - improved the navbar style
* [x] - improved the main page
* [x] - re-organized content structure
* [x] - added tutorial pages: `spark_with_zeppelin.md`, `python_with_zeppelin.md`, `sql_with_zeppelin.md` for overview
* [x] - added `multi_user_support.md` page to provide overview
* [x] - added the empty `interpreter_binding_mode` page. This will be handed in the different issue: [ZEPPELIN-2582](https://issues.apache.org/jira/browse/ZEPPELIN-2582)
* [x] - added the empty `trouble_shooting` page. This can be filled in the following PRs.
* [x] - added the empty `useful_developer_tools` page. This can be filled in the following PRs.

### What is the Jira issue?

[ZEPPELIN-2596](https://issues.apache.org/jira/browse/ZEPPELIN-2596)

### How should this be tested?

1. checkout
2. `cd docs`
3. `bundle install` (make sure that you have ruby 2.1.0+ and bundle)
4. `bundle exec jekyll serve --watch`
5. open `localhost:4000`

#### better navbar: before
2596_before_nav

#### better navbar: after

2596_after_nav

#### improved main page: before

2596_before_main

#### improved main page: after

2596_after_main

#### organized content structure: before

2596_before_content

#### organized content structure: after

2596_after_content

### Questions:
* Does the licenses files need update? - NO
* Is there breaking changes for older versions? - NO
* Does this needs documentation? -  related with docs

### CODE REVIEW

**Comment by AhyoungRyu** [comment]
Cool!

**Comment by 1ambda** [comment]
Thanks! @AhyoungRyu :)

**Comment by soralee** [comment]
Thanks @1ambda :+1: 
Let me test it out ASAP!

**Comment by soralee** [comment]
Sorry for my late response
I quickly checked some navbar bug  about https://github.com/apache/zeppelin/pull/2381#issuecomment-305451690 and https://github.com/apache/zeppelin/pull/2381#issuecomment-305705051 in #2381. and this branch has same those bugs.
@1ambda, could you check those bugs?

**Comment by 1ambda** [comment]
Thanks for the review @soralee. Let me update.

**Comment by 1ambda** [comment]
Just rebased to resolve conflicts. 

- Use less shadow for nav (f925348)
- Fix sora's comment 3 (91780b8)
- Fix sora's comment 1 (b36eea4)
- regarding sora's comment 2, we can solve it in a different PR. Since it's not normal use case (extending browser size from mobile to desktop)

**Comment by 1ambda** [comment]
Ready to review.

**Comment by soralee** [comment]
1. changing starter sentence in overview page

how about change the following sentence instead of `Apache Zeppelin` in `Spark with zeppelin`, `SQL with zeppelin` and `Python with zeppelin`? I mean, this part.
```
For a brief overview of Apache Spark fundamentals with Apache Zeppelin, see the following guide:
```
but, I'm not sure if this sentence is good. if you have more better sentence, It would be great!

[before]
image

[after]
image

**Comment by soralee** [comment]
### 2. (bug) Navbar doesn't fold when I click other components. This is same bug in https://github.com/apache/zeppelin/pull/2401#issuecomment-307291921

navbar_bug



and I fixed as a commit : https://github.com/apache/zeppelin/pull/2401/commits/b2e0c63381838ff1d96b9311679628202698ac92

**Comment by soralee** [comment]
### 3. removing `zeppelin` word in first line in description.
image

**Comment by soralee** [comment]
### 4. (bug) didn't revert original color when I click a navbar icon and escape. 
it is same bug with https://github.com/apache/zeppelin/pull/2401#issuecomment-307058027 as well.
don t_revet_navbarcolor


And I fixed a commit  : https://github.com/apache/zeppelin/pull/2401/commits/762635d09c879233cd498440e9f8b36d2e8c6988
although the two contents are in one commit, I hope it would help you.

**Comment by 1ambda** [comment]
@soralee thanks for the review. I fixed few things according to your comment. and the commit u provided was really helpful. 

Regarding 1.2, it's kind of a quick link for users who want to go the page directly. 

Let's say 

- I am a mysql user but doesn't know zeppelin and 
- i click `sql_with_zeppelin.html` since I want to how can I use SQL in zeppelin
- in this case, the list can be helpful.

**Comment by soralee** [comment]
@1ambda, I mean removing `Apache Zeppelin` phrase in description. (Sorry for my confusing screenshot)
image

File: docs/quickstart/python_with_zeppelin.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+-->
+{% include JB/setup %}
+
+# Python support in Zeppelin 
+
+<div id="toc"></div>
+
+<br/>
+
+Apache Zeppelin
```
Comment (soralee) [inline docs/quickstart/python_with_zeppelin.md]: 5. how about changing the following sentence?
```
The following guides explain how to using Apache Zeppelin that enables you to write in Python:
```
What do you think?

**Comment by 1ambda** [comment]
@soralee then, we need to update sentences. since we want to remove **subject**.

File: docs/quickstart/python_with_zeppelin.md (hunk 1 above)
Comment (1ambda) [inline docs/quickstart/python_with_zeppelin.md]: fixed.

**Comment by 1ambda** [comment]
@soralee what about this?

image

**Comment by soralee** [comment]
Great! it looks better!
And @1ambda Thank you for accepting suggest and quickly updating as well!

**Comment by 1ambda** [comment]
Ready to be reviewed.

**Comment by 1ambda** [comment]
Ready to review.

**Comment by 1ambda** [comment]
Ready to review.

**Comment by jongyoul** [comment]
LGTM. Merging if there is no more discussion

**Comment by soralee** [comment]
@1ambda Sorry for late response.
except one more thing bug : 
don t_revet_navbarcolor

FYI : https://github.com/apache/zeppelin/pull/2401/commits/c645ddddee0762df7ca51ef22dd390d9770a3223, I hope It would be help to you :slightly_smiling_face:

**Comment by 1ambda** [comment]
@soralee Thanks for the review. I missed fixing that part.

fold

**Comment by soralee** [comment]
@1ambda, Thanks for quick fix. LGTM!

**Comment by khalidhuseynov** [comment]
merging if no more discussion

**Comment by khalidhuseynov** [comment]
@1ambda could you plz resolved conflicts here?

**Comment by 1ambda** [comment]
@khalidhuseynov merged and checked that newly added thing included in this PR. 

image

from https://github.com/apache/zeppelin/commit/c90a3c2e51f1f1853c179b42ce4b241c518d31e3

**Comment by khalidhuseynov** [comment]
@1ambda thanks, will merge into master then

**Pull request closed without merging**